pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -1559,10 +1635,47 @@ public void setBlob(@Positive int parameterIndex,
     setLong(parameterIndex, oid);
   }
 
+  private void setBlobBytea(@Positive int parameterIndex,
+      @Nullable InputStream inputStream)
+      throws SQLException {
+
+    if (inputStream == null) {
+      setNull(parameterIndex, Types.VARBINARY);
+      return;
+    }
+
+    byte [] b;
+
+    try {
+      // for JDK 9+ we could use b = is.readAllBytes(). However ...
+      ByteArrayOutputStream os = new ByteArrayOutputStream();
+      byte[] buffer = new byte[4096];
+      for (int len = inputStream.read(buffer); len != -1; len = inputStream.read(buffer)) {
+        os.write(buffer, 0, len);
+      }
+      b = os.toByteArray();
+    } catch (IOException ioe) {
+      throw new PSQLException(GT.tr("Provided InputStream failed."), PSQLState.UNEXPECTED_ERROR,
+          ioe);
+    }
+
+    if (b.length > PgBlobBytea.MAX_BYTES) {
+      // XXX throw exception
```

Comment:
same as above, please throw the exception

Reply:
will fix